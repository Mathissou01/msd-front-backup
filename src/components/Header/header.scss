@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.z-index" as index;

.c-Header__ContentCover {
  position: absolute;
  z-index: index.z(content-cover);
  width: 100vw;
  height: 100vh;
  background-color: colors.$expert-blue;
  opacity: 0;
  transition: opacity ease 0.1s;
  pointer-events: none;

  &_active {
    opacity: 0.4;
    pointer-events: auto;
  }
}
