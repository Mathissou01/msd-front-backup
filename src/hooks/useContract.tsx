import React, { useContext } from "react";
import { ContractEntity } from "../graphql/codegen/generated-types";

export interface IContractContext {
  contract: ContractEntity;
  contractId: `${number}`;
  colors: { [key: string]: string };
}

export const ContractContext = React.createContext<IContractContext>({
  contract: {},
  contractId: "0",
  colors: {},
});

export const useContract = () => useContext(ContractContext);
