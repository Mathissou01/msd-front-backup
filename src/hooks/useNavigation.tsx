import React, { useContext } from "react";

export enum ENavigationPages {
  "HomeLink" = "/",
  // Editorial
  "ComponentLinksNews" = "/actualites-evenements",
  "ComponentLinksQuizzes" = "/quiz",
  "ComponentLinksTips" = "/astuces",
  "ComponentLinksFrees" = "/contenu-libre", // Path is concatenated with name as normalized string
  "ComponentLinksContactUs" = "/contact",
  "ComponentLinksMyWasteCounter" = "/mon-compteur-dechets",
  // Services
  "ComponentLinksRecyclingGuide" = "/service/guide-tri",
  "ComponentLinksDropOffMap" = "/service/carte",
  "ComponentLinksPickUpDay" = "/service/collecte-adresse",
  "ComponentLinksRequest" = "/service/demandes",
  // Misc
  "ComponentLinksExternal" = "/", // TODO: handle external links
}

export interface INavigationContext {
  currentPage: ENavigationPages | string;
  setCurrentPage: (pageName: ENavigationPages | string) => void;
}

export const NavigationContext = React.createContext<INavigationContext>({
  currentPage: ENavigationPages.HomeLink,
  setCurrentPage: () => null,
});

export const useNavigation = () => useContext(NavigationContext);
