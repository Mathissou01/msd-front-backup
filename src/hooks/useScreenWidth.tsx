import React, { useState, useEffect, useMemo, useContext } from "react";

const isWindow = typeof window !== "undefined";
const getRawDesktopBreakpoint = () =>
  isWindow
    ? Number.parseInt(
        getComputedStyle(document.documentElement).getPropertyValue(
          "--breakpoint-laptop",
        ),
      )
    : null;
const getDesktopBreakpoint = () => {
  const rawDesktopBreakpoint = getRawDesktopBreakpoint();
  return rawDesktopBreakpoint !== null && !isNaN(rawDesktopBreakpoint)
    ? rawDesktopBreakpoint
    : 1440;
};
const getIsDesktop = (width: number | undefined) => {
  const desktopBreakpoint = getDesktopBreakpoint();
  return width && desktopBreakpoint ? width >= desktopBreakpoint : undefined;
};

export interface IScreenWidth {
  windowWidth?: number;
  isDesktop?: boolean;
}

export default function useScreenWidth(): IScreenWidth {
  const getWidth = (): number | undefined =>
    isWindow ? window.innerWidth : undefined;
  const [windowWidth, setWindowWidth] = useState<number | undefined>(
    getWidth(),
  );
  const [isDesktop, setIsDesktop] = useState<boolean | undefined>(undefined);

  useEffect(() => {
    if (isWindow) {
      const resize = () => {
        const width = getWidth();
        setWindowWidth(width);
        setIsDesktop(getIsDesktop(width));
      };
      resize();
      window.addEventListener("resize", resize);
      return () => window.removeEventListener("resize", resize);
    }
  }, []);

  return useMemo(() => ({ windowWidth, isDesktop }), [windowWidth, isDesktop]);
}

export const IsDesktopContext = React.createContext<boolean | undefined>(
  undefined,
);

export const useIsDesktopContext = () => useContext(IsDesktopContext);
