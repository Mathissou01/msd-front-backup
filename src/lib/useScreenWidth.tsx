import { useState, useEffect } from "react";

export default function useScreenWidth() {
  const [windowWidth, setWindowWidth] = useState<number | null>(null);
  const [isDesktop, setIsDesktop] = useState<boolean>(false);
  const isWindow = typeof window !== "undefined";
  const getWidth = () => (isWindow ? window.innerWidth : windowWidth);
  const getDesktopBreakpoint = () =>
    isWindow
      ? Number.parseInt(
          getComputedStyle(document.documentElement).getPropertyValue(
            "--breakpoint-desktop",
          ),
        )
      : null;
  const resize = () => {
    const width = getWidth();
    const rawDesktopBreakpoint = getDesktopBreakpoint();
    // Set a default value of 1440 for the desktop breakpoint for Jest
    const desktopBreakpoint =
      rawDesktopBreakpoint !== null && !isNaN(rawDesktopBreakpoint)
        ? rawDesktopBreakpoint
        : 1440;
    setWindowWidth(width);
    setIsDesktop(
      width && desktopBreakpoint ? width >= desktopBreakpoint : false,
    );
  };

  useEffect(() => {
    if (isWindow) {
      resize();
      window.addEventListener("resize", resize);
      return () => window.removeEventListener("resize", resize);
    }
    //eslint-disable-next-line
  }, [isWindow]);

  return { windowWidth, isDesktop };
}
