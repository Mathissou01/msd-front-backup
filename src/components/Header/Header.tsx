import classNames from "classnames";
import Head from "next/head";
import { useState } from "react";
import { useIsDesktopContext } from "../../hooks/useScreenWidth";
import HeaderTopBar from "./HeaderTopBar/HeaderTopBar";
import HeaderSideBar from "./HeaderSideBar/HeaderSideBar";
import "./header.scss";
import { makePublicAssetPath } from "../../lib/utilities";

export default function Header() {
  const [sidebarExpanded, setSidebarExpanded] = useState(false);
  const isDesktop = useIsDesktopContext();
  const coverClassNames = classNames("c-Header__ContentCover", {
    "c-Header__ContentCover_active": sidebarExpanded && !isDesktop,
  });

  return (
    <>
      <Head>
        <title>MSD-FRONT</title>
        <meta name="description" content="wip" />
        <link rel="icon" href={makePublicAssetPath("/favicon.ico")} />
      </Head>
      <HeaderTopBar
        isMenuOpen={sidebarExpanded}
        isDesktopMode={isDesktop}
        handleClick={(e) => setSidebarExpanded(e)}
      />
      {isDesktop && <HeaderSideBar />}
      <div
        className={coverClassNames}
        aria-hidden={true}
        data-testid="content-cover"
      />
    </>
  );
}
