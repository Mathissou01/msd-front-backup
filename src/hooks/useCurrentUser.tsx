import React, {
  Dispatch,
  SetStateAction,
  createContext,
  useContext,
  useEffect,
  useState,
} from "react";
import { User } from "../lib/user";

interface UserContextValues {
  currentUser: User | null;
  setCurrentUser: Dispatch<SetStateAction<User | null>>;
  login: () => void;
  logout: () => void;
}

export const useCurrentUser = (): UserContextValues => {
  const context = useContext(UserContext);
  if (!context) {
    throw new Error();
  }
  return context;
};

const UserContext = createContext<UserContextValues>({} as UserContextValues);

export const UserProvider = ({ children }: { children: React.ReactNode }) => {
  const [currentUser, setCurrentUser] = useState<User | null>(null);

  const getUser = async () => {
    try {
      const storedUser = localStorage.getItem("user");
      if (storedUser && storedUser !== "null") {
        setCurrentUser(JSON.parse(storedUser));
      }
    } catch (error) {
      console.error(error);
    }
  };

  const login = () => {
    const user = {
      id: "1",
      email: "john.doe@gmail.com",
      firstname: "John",
      lastname: "Doe",
      activeCounter: true,
      householdSize: 1,
      userType: "particular",
      dwellingType: "house",
      addressLabel: "1 rue de la paix, 75000 Paris",
    };
    setCurrentUser(user);
    localStorage.setItem("user", JSON.stringify(user));
  };

  const logout = () => {
    setCurrentUser(null);
    localStorage.removeItem("user");
  };

  useEffect(() => {
    getUser();
  }, []);

  useEffect(() => {
    localStorage.setItem("user", JSON.stringify(currentUser));
  }, [currentUser]);

  const contextValue: UserContextValues = {
    currentUser,
    setCurrentUser,
    login,
    logout,
  };

  return (
    <UserContext.Provider value={contextValue}>{children}</UserContext.Provider>
  );
};

export default UserContext;
