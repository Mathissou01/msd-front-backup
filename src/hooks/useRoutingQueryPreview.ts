import { useRouter } from "next/router";
import { TPreviewTypes } from "../lib/preview";

export function useRoutingQueryPreview<T extends string | TPreviewTypes>(
  queryParam: string,
  validateQueryParam: (param: string) => boolean,
): T | undefined {
  const router = useRouter();
  const query = router.query[queryParam];
  if (query?.toString()) {
    if (!validateQueryParam(query.toString())) {
      void router.push("/404");
      return undefined;
    }
    return query?.toString() as T;
  } else if (typeof window === undefined) {
    void router.push("/404");
  }

  return undefined;
}
