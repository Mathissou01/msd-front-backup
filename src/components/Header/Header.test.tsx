import { fireEvent, render, screen } from "@testing-library/react";
import { IsDesktopContext } from "../../hooks/useScreenWidth";
import Header from "./Header";

jest.mock("./Navigation/NavigationList/NavigationList");

jest.mock("next/router", () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {},
  }),
}));

jest.mock("../../graphql/codegen/generated-types", () => ({
  useGetCitiesInformationsLazyQuery: jest.fn().mockReturnValue(["test"]),
}));

describe("Header", () => {
  it("renders in mobile mode", async () => {
    const { container } = render(
      <IsDesktopContext.Provider value={false}>
        <Header />
      </IsDesktopContext.Provider>,
    );
    const topBar = screen.getByTestId("top-bar");
    const burgerMenuButton = screen.queryByTestId("burger-menu-button");
    const burgerMenu = screen.getByTestId("burger-menu");
    const sideBar = screen.queryByTestId("side-bar");
    expect(topBar).toBeInTheDocument();
    expect(burgerMenuButton).toBeInTheDocument();
    expect(burgerMenu).toBeInTheDocument();
    expect(sideBar).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("opens the burger menu and enables the content cover when burger button is clicked", () => {
    render(
      <IsDesktopContext.Provider value={false}>
        <Header />
      </IsDesktopContext.Provider>,
    );
    const burgerMenuButton = screen.getByTestId("burger-menu-button");
    const contentCover = screen.getByTestId("content-cover");
    expect(burgerMenuButton).toBeInTheDocument();
    expect(contentCover).toBeInTheDocument();
    expect(contentCover).not.toHaveClass("c-Header__ContentCover_active");
    expect(burgerMenuButton).not.toHaveClass(
      "c-HeaderTopBar__BurgerButton_expanded",
    );
    fireEvent.click(burgerMenuButton);
    expect(contentCover).toHaveClass("c-Header__ContentCover_active");
    expect(burgerMenuButton).toHaveClass(
      "c-HeaderTopBar__BurgerButton_expanded",
    );
    fireEvent.click(burgerMenuButton);
    expect(contentCover).not.toHaveClass("c-Header__ContentCover_active");
    expect(burgerMenuButton).not.toHaveClass(
      "c-HeaderTopBar__BurgerButton_expanded",
    );
  });
  it("renders in desktop mode", () => {
    const { container } = render(
      <IsDesktopContext.Provider value={true}>
        <Header />
      </IsDesktopContext.Provider>,
    );
    const topBar = screen.getByTestId("top-bar");
    const sideBar = screen.getByTestId("side-bar");
    const burgerMenuButton = screen.queryByTestId("burger-menu-button");
    const burgerMenu = screen.queryByTestId("burger-menu");
    expect(sideBar).toBeInTheDocument();
    expect(topBar).toBeInTheDocument();
    expect(burgerMenuButton).not.toBeInTheDocument();
    expect(burgerMenu).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
