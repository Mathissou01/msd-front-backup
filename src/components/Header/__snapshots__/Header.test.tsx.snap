// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders in desktop mode 1`] = `
<div>
  <header
    class="c-HeaderTopBar"
    data-testid="top-bar"
    role="banner"
  >
    <div
      class="c-HeaderTopBar__Item c-HeaderTopBar__Logo"
    >
      <a
        class="c-LogoCommunity"
        data-testid="logo-community"
        href="/"
      >
        <div />
      </a>
    </div>
    <ul
      class="c-HeaderTopBar__List"
    >
      <li
        class="c-HeaderTopBar__Item"
      >
        <div
          class="c-HeaderLinkSelector"
        >
          <span>
            Montauban
          </span>
          <div />
        </div>
      </li>
      <div
        aria-hidden="true"
        class="c-HeaderTopBar__Filler"
      />
      <li
        class="c-HeaderTopBar__Item"
      >
        <div
          class="c-HeaderTopBar__SearchBar"
        >
          <div
            class="c-HeaderLinkSearch"
          >
            <input
              class="c-HeaderLinkSearch__Input"
              placeholder="Rechercher sur le site"
              type="text"
            />
            <button
              class="c-CommonButton c-CommonButton_tertiary c-CommonButton_fontSmall"
              type="button"
            >
              <div
                class="c-CommonButton__Picto c-CommonButton__Picto_search"
              />
            </button>
          </div>
        </div>
      </li>
      <li
        class="c-HeaderTopBar__Item"
      >
        <a
          class="c-NavigationLink c-NavigationLink_desktop"
          href="/"
        >
          <div
            class="c-NavigationLink__Box"
          >
            <div
              class="c-NavigationLink__Picto"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height="24"
                loading="lazy"
                src="/images/pictos/localisation.svg"
                srcset="/images/pictos/localisation.svg 1x, /images/pictos/localisation.svg 2x"
                style="color: transparent;"
                width="24"
              />
            </div>
          </div>
          <span
            class="c-NavigationLink__Label"
          />
        </a>
      </li>
      <div
        aria-hidden="true"
        class="c-HeaderTopBar__Separator"
      />
      <li
        class="c-HeaderTopBar__Item"
      >
        <a
          class="c-NavigationLink c-NavigationLink_desktop"
          href="/"
        >
          <div
            class="c-NavigationLink__Box"
          >
            <div
              class="c-NavigationLink__Picto"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height="24"
                loading="lazy"
                src="/images/pictos/calendar.svg"
                srcset="/images/pictos/calendar.svg 1x, /images/pictos/calendar.svg 2x"
                style="color: transparent;"
                width="24"
              />
            </div>
          </div>
          <span
            class="c-NavigationLink__Label"
          />
        </a>
      </li>
      <div
        aria-hidden="true"
        class="c-HeaderTopBar__Separator"
      />
      <li
        class="c-HeaderTopBar__Item"
      >
        <a
          class="c-NavigationLink c-NavigationLink_desktop"
          href="/"
        >
          <div
            class="c-NavigationLink__Box"
          >
            <div
              class="c-NavigationLink__Picto"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height="24"
                loading="lazy"
                src="/images/pictos/notify.svg"
                srcset="/images/pictos/notify.svg 1x, /images/pictos/notify.svg 2x"
                style="color: transparent;"
                width="24"
              />
            </div>
          </div>
          <span
            class="c-NavigationLink__Label"
          />
        </a>
      </li>
      <div
        aria-hidden="true"
        class="c-HeaderTopBar__Separator"
      />
      <li
        class="c-HeaderTopBar__Item"
      >
        <a
          class="c-NavigationLink c-NavigationLink_desktop"
          href="/"
        >
          <div
            class="c-NavigationLink__Box"
          >
            <div
              class="c-NavigationLink__Picto"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height="24"
                loading="lazy"
                src="/images/pictos/contact.svg"
                srcset="/images/pictos/contact.svg 1x, /images/pictos/contact.svg 2x"
                style="color: transparent;"
                width="24"
              />
            </div>
          </div>
          <span
            class="c-NavigationLink__Label"
          />
        </a>
      </li>
      <div
        aria-hidden="true"
        class="c-HeaderTopBar__Separator"
      />
      <li
        class="c-HeaderTopBar__Item"
      >
        <div
          class="c-HeaderLinkProfile"
        >
          <span>
            Particulier
          </span>
          <div />
        </div>
      </li>
      <div
        aria-hidden="true"
        class="c-HeaderTopBar__Separator"
      />
      <li
        class="c-HeaderTopBar__Item"
      >
        <a
          class="c-HeaderLinkUser"
          href="/"
        >
          <div
            class="c-HeaderLinkUser__Circle"
          >
            <div />
          </div>
        </a>
      </li>
    </ul>
  </header>
  <nav
    class="c-HeaderSideBar"
    data-testid="side-bar"
  >
    <a
      class="c-MSDLogo"
      href="https://www.monservicedechets.com/"
    >
      <div />
    </a>
  </nav>
  <div
    aria-hidden="true"
    class="c-Header__ContentCover"
    data-testid="content-cover"
  />
</div>
`;

exports[`Header renders in mobile mode 1`] = `
<div>
  <header
    class="c-HeaderTopBar"
    data-testid="top-bar"
    role="banner"
  >
    <button
      aria-expanded="false"
      class="c-HeaderTopBar__BurgerButton"
      data-testid="burger-menu-button"
      type="button"
    />
    <div
      class="c-HeaderTopBar__Item c-HeaderTopBar__Logo"
    >
      <a
        class="c-LogoCommunity"
        data-testid="logo-community"
        href="/"
      >
        <div />
      </a>
    </div>
    <ul
      class="c-HeaderTopBar__List"
    >
      <li
        class="c-HeaderTopBar__Item c-HeaderTopBar__User"
      >
        <a
          class="c-HeaderLinkUser"
          href="/"
        >
          <div
            class="c-HeaderLinkUser__Circle"
          >
            <div />
          </div>
        </a>
      </li>
      <li
        class="c-HeaderTopBar__Item c-HeaderTopBar__Search"
      >
        <a
          class="c-NavigationLink c-NavigationLink_desktop"
          href="/"
        >
          <div
            class="c-NavigationLink__Box"
          >
            <div
              class="c-NavigationLink__Picto"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height="24"
                loading="lazy"
                src="/images/pictos/search.svg"
                srcset="/images/pictos/search.svg 1x, /images/pictos/search.svg 2x"
                style="color: transparent;"
                width="24"
              />
            </div>
          </div>
          <span
            class="c-NavigationLink__Label"
          />
        </a>
      </li>
    </ul>
  </header>
  <nav
    class="c-HeaderBurgerMenu"
    data-testid="burger-menu"
  >
    <ul
      class="c-HeaderBurgerMenu__List"
    >
      <li
        class="c-HeaderBurgerMenu__Item"
      >
        <h2>
          Ma commune
        </h2>
        <div
          class="c-HeaderLinkSelector"
        >
          <span>
            Montauban
          </span>
          <div />
        </div>
      </li>
      <li
        class="c-HeaderBurgerMenu__Item"
      >
        <div
          class="c-HeaderBurgerMenu__User"
        >
          <span>
            Je suis
          </span>
          <div
            class="c-HeaderLinkProfile"
          >
            <span>
              Particulier
            </span>
            <div />
          </div>
        </div>
      </li>
      <li
        class="c-HeaderBurgerMenu__Item c-HeaderBurgerMenu__Item_grow"
      >
        <nav
          class="c-HeaderBurgerMenu__Container"
          role="navigation"
        >
          <h2>
            Mes services
          </h2>
        </nav>
      </li>
      <li
        class="c-HeaderBurgerMenu__Item c-HeaderBurgerMenu__Item_grow"
      >
        <nav
          class="c-HeaderBurgerMenu__Container"
          role="navigation"
        >
          <a
            class="c-NavigationLink c-NavigationLink_mobile"
            href="/"
          >
            <div
              class="c-NavigationLink__Box"
            >
              <div
                class="c-NavigationLink__Picto"
              >
                <img
                  alt=""
                  data-nimg="1"
                  decoding="async"
                  height="24"
                  loading="lazy"
                  src="/images/pictos/localisation.svg"
                  srcset="/images/pictos/localisation.svg 1x, /images/pictos/localisation.svg 2x"
                  style="color: transparent;"
                  width="24"
                />
              </div>
            </div>
            <span
              class="c-NavigationLink__Label"
            >
              Mon adresse
            </span>
          </a>
          <a
            class="c-NavigationLink c-NavigationLink_mobile"
            href="/"
          >
            <div
              class="c-NavigationLink__Box"
            >
              <div
                class="c-NavigationLink__Picto"
              >
                <img
                  alt=""
                  data-nimg="1"
                  decoding="async"
                  height="24"
                  loading="lazy"
                  src="/images/pictos/calendar.svg"
                  srcset="/images/pictos/calendar.svg 1x, /images/pictos/calendar.svg 2x"
                  style="color: transparent;"
                  width="24"
                />
              </div>
            </div>
            <span
              class="c-NavigationLink__Label"
            >
              Calendrier
            </span>
          </a>
          <a
            class="c-NavigationLink c-NavigationLink_mobile"
            href="/"
          >
            <div
              class="c-NavigationLink__Box"
            >
              <div
                class="c-NavigationLink__Picto"
              >
                <img
                  alt=""
                  data-nimg="1"
                  decoding="async"
                  height="24"
                  loading="lazy"
                  src="/images/pictos/notify.svg"
                  srcset="/images/pictos/notify.svg 1x, /images/pictos/notify.svg 2x"
                  style="color: transparent;"
                  width="24"
                />
              </div>
            </div>
            <span
              class="c-NavigationLink__Label"
            >
              Alertes
            </span>
          </a>
          <a
            class="c-NavigationLink c-NavigationLink_mobile"
            href="/"
          >
            <div
              class="c-NavigationLink__Box"
            >
              <div
                class="c-NavigationLink__Picto"
              >
                <img
                  alt=""
                  data-nimg="1"
                  decoding="async"
                  height="24"
                  loading="lazy"
                  src="/images/pictos/contact.svg"
                  srcset="/images/pictos/contact.svg 1x, /images/pictos/contact.svg 2x"
                  style="color: transparent;"
                  width="24"
                />
              </div>
            </div>
            <span
              class="c-NavigationLink__Label"
            >
              Contact
            </span>
          </a>
        </nav>
      </li>
      <li
        class="c-HeaderBurgerMenu__Item"
      >
        <a
          class="c-MSDLogo"
          href="https://www.monservicedechets.com/"
        >
          <div />
        </a>
      </li>
    </ul>
  </nav>
  <div
    aria-hidden="true"
    class="c-Header__ContentCover"
    data-testid="content-cover"
  />
</div>
`;
